app/scripts/controllers/permissions/methodMiddleware.js

```diff
@@ -149,15 +153,15 @@ module.exports = function createRequestMiddleware ({
         // install plugins, if any
 
         // get the names of the approved plugins
-        const pluginPrefixRegex = new RegExp(`^${PLUGIN_PREFIX}`)
         const requestedPlugins = result.permissions
           // requestPermissions returns all permissions for the domain,
-          // so we're filtering out non-plugin and existing permissions
+          // so we're filtering out non-plugin and preexisting permissions
           .filter(p => (
             p.parentCapability.startsWith(PLUGIN_PREFIX) &&
             p.parentCapability in requestedPermissions
           ))
-          .map(p => p.parentCapability.replace(pluginPrefixRegex, ''))
+          // convert from namespaced permissions to plugin names
+          .map(p => p.parentCapability.replace(PLUGIN_PREFIX_REGEX, ''))
           .reduce((acc, pluginName) => {
             acc[pluginName] = {}
```

Comment:
If the plugin permissions included some options, would we want to preserve them here, rather than initializing a new object? Not a blocker, since we don't yet have plugin install options.